docs/scripts/util/runArrayPromiseInOrder.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Asynchronous array loops
+ * More details: https://www.30secondsofcode.org/articles/s/javascript-async-array-loops#for-loops
+ * @param {array} arr - array to iterate
+ * @param {function} fn - callback function
+ */
+export async function runArrayPromiseInOrder(arr: unknown[], fn) {
```

Comment:
Can we better document why we need this util instead of using `Promise.all`?

Reply:
`Promise.all` only cares if all the promises completed, but `runArrayPromiseInOrder` waits for the first promise to complete before running the next one.

I can add some comments

Reply:
Actually, the link in the comment has already explained that 😂

Reply:
Here's a better article discussing this problem https://www.nodejsdesignpatterns.com/blog/node-js-race-conditions/

TL;DR: 
Here's the issue:

![image](https://user-images.githubusercontent.com/11983489/222601354-eb4a7e78-3768-453a-aeb1-3b0920e65c06.png)

Here's the solution:

![image](https://user-images.githubusercontent.com/11983489/222601488-d3d1388f-357c-458c-89b4-5c98cee15546.png)

Reply:
Sorry if i didn't make this transparent in the initial comment, but to make it easier for someone working on this a year from now we should make it clear in our comment why we needed this function on the first place, IMO linking to a blog post does not explain why we need this in our code. 

Missed this earlier, but a couple other things:
- consider renaming to `runOrderedPromises`
- the function signature should expect an array of promises right?

Reply:
Is having these sequential and ordered a requirement though? Executing promises in parallel seems more valuable imo, would make the test run faster as well. If reaching node eventemitter limits is the concerns we can try utils such as `p-limit` to limit the number of promises that run concurrently.

Reply:
[p-limit](https://www.npmjs.com/package/p-limit) looks interesting. Let me try if we can use a limit greater than 1 🤔

Reply:
Even though checking the links concurrently may save us some time, it might be hard to debug if there's a failure in the middle. For example this case 
https://github.com/0618/amplify-ui/actions/runs/4328294627/jobs/7557919861 since we run the links in order, we can easily locate the page and link

Reply:
Thanks for suggesting [PromisePool](https://www.npmjs.com/package/@supercharge/promise-pool), @ioanabrooks ! The time of the test is significantly reduced from about 480s on Github to 170s. And only takes 60s on my cloud desktop 🎉 

Code is updated.